### ISSUE TITLE
Can somebody take over? This project seems abandoned by henrywhitaker3

### ISSUE THREAD

**Issue by Patrick010** [henrywhitaker3/Speedtest-Tracker#1013]
Unfortunately no activity from the maintainer.
The fact that there are 129 active issues proves that there is an active usergroup.
Anybody willing to take over developing Speedtest?

**Comment by alexjustesen** [comment]
Just stumbled across this project. I'm thinking of making a fork and getting it updated along with addressing some of those issues.

**Comment by yllekz** [comment]
Yeah I've noticed there hasn't been any substantial improvement or releases in a while. There's still a few major bugs on the main dashboard (graphs not populating beyond 3 days ago despite there being data). Hoping more contributors get elevated.

**Comment by alexjustesen** [comment]
I've got two clients I'm finishing up work on but I plan to pick this up as a fork with a buddy once that is done

**Comment by zackmuc** [comment]
For me it does not work since one week, how about you?

**Comment by alexjustesen** [comment]
@zackmuc mines been running fine (using the docker image on unraid). Are you getting any errors in the logs?

**Comment by zackmuc** [comment]
get: ==============================================================================
To accept the message please run speedtest interactively or use the following: 
    speedtest --accept-gdpr
==============================================================================
Ookla collects certain data through Speedtest that may be considered
personally identifiable, such as your IP address, unique device
identifiers or location. Ookla believes it has a legitimate interest
to share this data with internet providers, hardware manufacturers and
industry regulators to help them understand and create a better and
faster internet. For further information including how the data may be
shared, where the data may be transferred and Ookla's contact details,
please see our Privacy Policy at:
       http://www.speedtest.net/privacy
==============================================================================
To accept the message please run speedtest interactively or use the following: 
 but where am i doing this

**Comment by zackmuc** [comment]
the license  switch does not work anymore somehow

**Comment by zackmuc** [comment]
and i need the arm64 version :(

**Comment by alexjustesen** [comment]
Thanks, I'll add this to my notes. I have a couple rpi's so I can test arm64 when I get there. No start date yet.

**Comment by zackmuc** [comment]
got the solution: 

if you docker exec as root and run php /config/www/artisan speedtest:eula - it'll start working again.

**Comment by jpslv** [comment]
It would be great if someone continued this excellent work.

**Comment by fmantz** [comment]
I just found this project by accident ...I developed something similar myself some months ago (LOL) when I had frequently problems with my internet connection. I run it on a pi even I provide on dockerhub an image for amd64 only (for the NAS of a friend). It has less features (!) ... but maybe it is interesting for someone. I use an unofficial speedtest.net client that I forked and modified. It generates a statically web page by a small rust program scheduled by a cron job. Since the program I forked does not require to accept a license, I did not include such a check (hope I am not in legal trouble, please check https://www.speedtest.net/about/privacy). You can find it here https://github.com/fmantz/SpeedTracker.  (can be used on own risk / was also a case study for me to learn some RUST). The link contains also a screen shot. It was originally only for my personal use ... so do not expect too much ;-)

**Comment by alexjustesen** [comment]
Started playing around to make sure this was viable from a CLI standpoint. Tests are successfully running and being logged to a database so we'll call that a win for the evening. In the next couple of days I'll open a roadmap / suggestions intake so I can get things in the pipeline.

### Personal shortlist of features
- accept the license automatically and get rid of those pesky license errors
- be able to run tests adhoc or on a schedule
- test against "closest" server or a defined server
- scheduled task monitoring using https://ohdear.app, https://thenping.me or https://healthchecks.io
- support storing speedtest results to different databases like sqlite, mariadb/mysql, postgresql and [influxdb](https://aschmelyun.com/blog/storing-time-series-data-in-influxdb-with-laravel/)
- send notifications when a test runs success/failure or when it's outside an ideal threshold
- support notification channels like mail, slack, discord, telegram or really anything that makes sense from [this list](https://laravel-notification-channels.com/about/)
- K.I.S.S.

### Results preview
![image](https://user-images.githubusercontent.com/1144087/185282003-26859089-ff4c-47bf-9c86-e25debe6ce89.png)

**Comment by alexjustesen** [comment]
To those following along, I've setup a roadmap where you can track in-progress work and suggest features. https://speedtest-tracker-roadmap.alexjustesen.dev/

**Comment by jpslv** [comment]
@alexjustesen, I'm getting "Error code 1020".

**Comment by alexjustesen** [comment]
> @alexjustesen, I'm getting "Error code 1020".

Try one more time, forgot to take off the domain's geo restrictions

**Comment by Patrick010** [comment]
> To those following along, I've setup a roadmap where you can track in-progress work and suggest features. https://speedtest-tracker-roadmap.alexjustesen.dev/

Sweet! Thanks

**Comment by alexjustesen** [comment]
@Patrick010 it's a short list right now so if you feel as though features are missing please create an item.

**Comment by Patrick010** [comment]
@alexjustesen Not sure what your goals are. Are going to continue Henry's project, or something new or alike? Where will you publish your builds and will there be an official fork on Github? I've looked at the link you provided, looks like your private development environment.

**Comment by alexjustesen** [comment]
@Patrick010 my goal is to do a complete refactor the application and start from scratch. Laravel (the framework used in this project) has evolved and so its easier to just start from a fresh codebase. 

My baseline for usability is to port over all the same features and dashboard views that currently exist. After that is done I'll add new functionality. Priority for me will be the API so the data is accessible to anything.

Project repo is going to be private until something usable is out there and then I'll make it public.

Oh and one more thing... I found this because of the Unraid app so I'll be supporting that too as that's where I'm running it too.

**Comment by Patrick010** [comment]
Ok nice. I'm running in on in a Docker container on a Synology DS920+
Lets see how this evolves :)

**Comment by alexjustesen** [comment]
I got one of those too so I'll add that to my tests, also going to include a Raspberry Pi so it'll work on arm64 as well.

**Comment by luk6173** [comment]
> Project repo is going to be private until something usable is out there and then I'll make it public.

if you need some testers, let me know :)

**Comment by BananaHammock743** [comment]
> > Project repo is going to be private until something usable is out there and then I'll make it public.
> 
> if you need some testers, let me know :)

Happy to be a tester 👌

**Comment by alexjustesen** [comment]
@c1xx @BananaHammock743 (nice username 🤣) if you (or anyone who wants to be in the testing group) can do me a favor and comment on the project issue below that'd be great.

[I want to be a test subject](https://speedtest-tracker-roadmap.alexjustesen.dev/projects/3-public-beta/items/10-i-want-to-be-a-tester)

**Comment by Patrick010** [comment]
> @c1xx @BananaHammock743 (nice username 🤣) if you (or anyone who wants to be in the testing group) can do me a favor and comment on the project issue below that'd be great.
> 
> [I want to be a test subject](https://speedtest-tracker-roadmap.alexjustesen.dev/projects/3-public-beta/items/10-i-want-to-be-a-tester)

Let me know whenever you need Docker testers :)

https://github.com/Patrick010

**Comment by Charlie67j** [comment]
Indeed. I run the speed test in a container running on a PXE host and another one, also in a docker container on an Ubuntu x86 host. I can help with a little testing as well.

**Comment by alexjustesen** [comment]
For those following along... Progress has been made! 

At this point I've got everything running in a "headless mode" to ensure the scheduler, queue and speedtests work automatically.

Should you want to become a tester or follow the progress you can do so here: https://speedtest-tracker-roadmap.alexjustesen.dev/

**Comment by michealespinola** [comment]
I don't understand the purpose of the external site. Why not just fork it here on Github?

**Comment by Patrick010** [comment]
@michealespinola I sort of thought the same, but it's @alexjustesen's project, so that's up to him.

**Comment by alexjustesen** [comment]
> @michealespinola I sort of thought the same, but it's @alexjustesen's project, so that's up to him.

@michealespinola I'll have a public repo soon, I just like having something to share before doing so.

I also explained it in a road map comment but I decided not to fork the current repo as the Laravel framework has come a long way since this version so a fresh start was easier.

**Comment by alexjustesen** [comment]
@michealespinola f-it... I decided to build in public if anyone wants to follow along here's the repo. Fair warning it's a work in progress in every sense of the term.

https://github.com/alexjustesen/speedtest-tracker

**Comment by Patrick010** [comment]
@alexjustesen Don't feel pushed in any way. It's just that 'people' are used to github and know their way. Especially if its a fork or adoption of an already published thing. Thanks for your efforts, they're highly appreciated :) 👍

**Comment by alexjustesen** [comment]
@Patrick010 I was already moving in that direction as private repos w/ packages are a pain and cause extra steps. Benefit is now everyone can have a voice sooner. I may not do it but at least we can get ideas and feedback into the pipeline early

**Comment by alexjustesen** [comment]
If anyone has a sample of data they can provide out of InfluxDB that would be really helpful. https://github.com/alexjustesen/speedtest-tracker/issues/12

**Comment by Patrick010** [comment]
@alexjustesen What is it you're looking for exactly? filename?

**Comment by alexjustesen** [comment]
@Patrick010 more so the data and the format it's provided from the tracker to influxdb. If you can there is a discussion on the issue above

**Comment by alexjustesen** [comment]
I need the 2c of the community, if you got a couple seconds let me know if an API would be useful for you: https://github.com/alexjustesen/speedtest-tracker/discussions/22

**Comment by p3rfyct** [comment]
Does seem like there is active interest in this so I wanted to request a feature (not sure if someone is stepping up to own it going forward) a way to add an additional connection for testing (e.g. testing a vpn tunnel performance) concurrently through the same interface

**Comment by michealespinola** [comment]
You're going to want to make any requests over at [alexjustesen/speedtest-tracker](https://github.com/alexjustesen/speedtest-tracker)
